wg7adapter/src/main/java/net/alex9849/arm/adapters/WG7Region.java

```diff
@@ -175,8 +176,41 @@ public boolean isPolygonal() {
         return region instanceof ProtectedPolygonalRegion;
     }
 
+    public static double calculatePoly2DRegionVolume(List<Vector2> polygonPoints) {
+        int numPoints = polygonPoints.size();
+        if (numPoints < 3) {
+            return 0D;
+        }
+
+        double volume = 0D;
+        Vector2 baseVector = polygonPoints.get(0);
+
+        for (int i = 1; i < numPoints - 1; i++) {
+            Vector2 vector1 = polygonPoints.get(i).subtract(baseVector);
+            Vector2 vector2 = polygonPoints.get(i + 1).subtract(baseVector);
+
+            double crossProduct = vector1.getX() * vector2.getZ() - vector1.getZ() * vector2.getX();
+            double triangleArea = crossProduct / 2F;
+            double sideLength = vector1.length();
+
+            volume += triangleArea * sideLength;
+        }
+
+        return Math.abs(volume);
+    }
+
     @Override
     public int getVolume() {
+        if (region.getPoints().size() > 4) {
```

Comment:
Polygonal regions can also have 4 points. The lines between them are just not all alligned to the minecraft blockgrid.